src/aap_eda/api/serializers/activation.py

```diff
@@ -11,18 +11,55 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
+import logging
+
+import yaml
 from rest_framework import serializers
 
+from aap_eda.api.constants import PG_NOTIFY_TEMPLATE_RULEBOOK_DATA
 from aap_eda.api.serializers.credential import CredentialSerializer
 from aap_eda.api.serializers.decision_environment import (
     DecisionEnvironmentRefSerializer,
 )
+from aap_eda.api.serializers.event_stream import EventStreamSerializer
 from aap_eda.api.serializers.project import (
     ExtraVarRefSerializer,
     ProjectRefSerializer,
 )
 from aap_eda.api.serializers.rulebook import RulebookRefSerializer
+from aap_eda.api.serializers.utils import (
+    substitute_extra_vars,
+    substitute_source_args,
+    swap_sources,
+)
 from aap_eda.core import models, validators
+from aap_eda.core.enums import ProcessParentType
+
+logger = logging.getLogger(__name__)
+
+
+def _updated_ruleset(validated_data):
+    sources_info = []
+
+    for event_stream_id in validated_data["event_streams"]:
+        event_stream = models.EventStream.objects.get(id=event_stream_id)
+
+        if event_stream.rulebook:
+            rulesets = yaml.safe_load(event_stream.rulebook.rulesets)
+        else:
+            rulesets = yaml.safe_load(PG_NOTIFY_TEMPLATE_RULEBOOK_DATA)
+        extra_vars = rulesets[0]["sources"][0]["extra_vars"]
+        extra_vars = substitute_extra_vars(
+            event_stream.__dict__, extra_vars, [], "password"
```

Comment:
@hsong-rh Can you put a TODO for the password once @bzwei has implemented the ansible-rulebook with valut support we will fill in the correct password and encrypt the data with it using ansible-vault encrypt feature.